components/layout_2020/fragments.rs

```diff
@@ -180,10 +178,33 @@ impl BoxFragment {
             border,
             margin,
             block_margins_collapsed_with_children,
-            scrollable_overflow,
+            scrollable_overflow: PhysicalRect::zero(),
```

Comment:
I’m not a fan of initializing here with known-incorrect data and rely on `calculate_scrollable_overflow` being called later at some point for its side effect. Why is that computation delayed? If it needs to be, what do you think of making this an `Option` and calling `unwrap` when the data is used, so we don’t accidentally use a non-initialized zero?

Reply:
The issue is that the BoxFragment is created and then positioned later. Does it make sense to modify `layout_block_level_children` to create and place `BoxFragment`s at the same time?

Reply:
Creation is separated from placement in `layout_block_level_children` so that the former (including recursion in a subtree) can be parallelized across siblings, while the latter needs to be sequential.

What if we defined `BoxFragment::scrollable_overflow` to be in a coordinate system where `self.content_rect.start_corner` is the origin, and the parent is responsible for doing the translation?

Reply:
I have done something like this in the latest branch. Now the final position of the scrollable overflow is calculated when it is used. This shouldn't have a large overhead. I've kept the calculation inside of the child since it knows about its position when the calculation is done.